src/Components/Feature/ToDoList.tsx

```diff
@@ -1,42 +1,93 @@
-import { useEffect, useState } from 'react';
+import { ChangeEvent, useEffect, useState } from 'react';
 import { IToDoResp, IToDoRespD } from '../../Shared/types'
 import Todo from './ToDo'
 import useFetch from '../Common/useFetch';
+import FilterBar from './FIlterBar';
 
 
 const ToDoList = () => {
-  const [todos, setTodos] = useState<IToDoResp[] | null>(null);
-
+  const [originalTodos, setOriginalFilteredTodos] = useState<IToDoResp[]>([]);
+  const [filteredTodos, setFilteredTodos] = useState<IToDoResp[]>([]);
+  const [searchTerm, setSearchTerm] = useState<string>('');
+  const [sortBy, setSortBy] = useState<string>('');
+  const [statusFilter, setStatusFilter] = useState<string>('');
   const { data, loading, error }:IToDoRespD = useFetch('https://jsonplaceholder.typicode.com/todos');
 
   useEffect(() => {
     if (data) {
-      setTodos(data);
+      setFilteredTodos(data);
+      setOriginalFilteredTodos(data)
     }
   }, [data]);
 
   const markCompleted = (id: number) => {
-    if (todos) {
-      const updatedTodos:IToDoResp[] | null = todos.map(todo =>
+    if (filteredTodos) {
+      const updatedTodos:IToDoResp[] | null = filteredTodos.map(todo =>
         todo.id === id ? { ...todo, completed: !todo.completed } : todo
       );
-      setTodos(updatedTodos);
+      setFilteredTodos(updatedTodos);
     }
   };
 
   const deleteTodo = (id: number) => {
-    if (todos) {
-      const updatedTodos = todos.filter(todo => todo.id !== id);
-      setTodos(updatedTodos);
+    if (filteredTodos) {
+      const updatedTodos = filteredTodos.filter(todo => todo.id !== id);
+      setFilteredTodos(updatedTodos);
     }
   };
 
+    // Handle search input
+    const handleSearch = (e: ChangeEvent<HTMLInputElement>) => {
```

Comment:
can you please optimize these handlers?
This handler will create every time when components rerenders